lib/subjects/selector.rb

```diff
@@ -131,7 +131,7 @@ def dequeue_ids(queue, sms_ids)
       if queue_user
         queue.dequeue_update(sms_ids)
       else
-        DequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
+        NonLoggedInDequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
       end
     rescue ActiveRecord::StaleObjectError
       DequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
```

Comment:
Because I just now had to think about whether this one should have been changed as well, should we move this rescue block so that the method reads:

```ruby
    def dequeue_ids(queue, sms_ids)
      if queue_user
        begin
          queue.dequeue_update(sms_ids)
        rescue ActiveRecord::StaleObjectError
          DequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
        end
      else
        NonLoggedInDequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
      end
    end
```

To make it clear that the DequeueSubjectQueueWorker is a fallback in case inline removal for a user queue didn't work the first try?